actor/v7action/service_instance.go

```diff
@@ -184,6 +184,58 @@ func (actor Actor) CreateManagedServiceInstance(params ManagedServiceInstancePar
 
 }
 
+type ServiceInstanceDeleteState int
+
+const (
+	ServiceInstanceUnknownState ServiceInstanceDeleteState = iota
+	ServiceInstanceDidNotExist
+	ServiceInstanceGone
+	ServiceInstanceDeleteInProgress
+)
+
+func (actor Actor) DeleteServiceInstance(serviceInstanceName, spaceGUID string, wait bool) (ServiceInstanceDeleteState, Warnings, error) {
+	var (
+		serviceInstance resources.ServiceInstance
+		jobURL          ccv3.JobURL
+	)
+
+	warnings, err := railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			serviceInstance, _, warnings, err = actor.CloudControllerClient.GetServiceInstanceByNameAndSpace(serviceInstanceName, spaceGUID)
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			if serviceInstance.GUID != "" {
+				jobURL, warnings, err = actor.CloudControllerClient.DeleteServiceInstance(serviceInstance.GUID)
+			}
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			switch {
+			case jobURL == "":
+				return
+			case wait:
+				return actor.CloudControllerClient.PollJob(jobURL)
+			default:
+				return actor.CloudControllerClient.PollJobForState(jobURL, constant.JobPolling)
+			}
+		},
+	)
```

Comment:
I like that `railway.Sequentially` offers the option to make the code more readable. I think if the functions are also extracted it would be even more readable.